### PR TITLE
Angr in GUI

### PR DESCRIPTION
**Pull request by EdwardLarson**
**Link to Related Issue(s)**
N/A

**Please describe the changes in your request.**
* Add an option to set Angr as the disassembler backend when starting the GUI
* Instead of Ghidra being the default backend if none is provided, no backend is used (no disassembly is possible, but this allows the GUI server to be run in an image/install that only has core OFRAK + frontend with no extra dependencies).
* Add a serializer class to "catch" the Angr analysis object so that it is not erroneously serialized when communicating with the frontend. This serializer doesn't actually correctly serialize the object, but instead always serializes it to `None` and raises a hard error on deserializing, as saving disassembler backend metadata is not currently supported.
* A few updates/improvements for the GUI development rules

**Anyone you think should look at this, specifically?**
@rbs-jacob

### CODE REVIEW

File: frontend/backend/ofrak_server.Dockerfile [hunk 1]
```diff
@@ -1,12 +1,11 @@
 FROM redballoonsecurity/ofrak/dev:latest
 
-COPY ofrak_server.py /ofrak_server.py
-COPY docs /docs
-COPY mkdocs.yml /mkdocs.yml
```
Comment (rbs-jacob) [inline frontend/backend/ofrak_server.Dockerfile:5]: Why remove the docs from the development Dockerfile?

Comment (rbs-jacob) [reply frontend/backend/ofrak_server.Dockerfile:5]: Moreover, if you remove the docs from the image, the `mkdocs serve` run in the entrypoint probably won't work?

Comment (EdwardLarson) [reply frontend/backend/ofrak_server.Dockerfile:5]: Clicking on a tag in the frontend takes me to the public ofrak.com docs anyway, so the dev server hosted docs are basically unused. If we were testing the documentation build here, we should have a dedicated server build for that (in the `docs` directory).

Comment (EdwardLarson) [reply frontend/backend/ofrak_server.Dockerfile:5]: And you're right about the mkdocs serve commend, I mean to remove that and could have sworn I had. Definitely removed now.

**Review by rbs-jacob — APPROVED** [review]

**Pull request merged**